pkg/attestation/crafter/runners/githubaction.go

```diff
@@ -16,16 +16,48 @@
 package runners
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/pkg/attestation/crafter/runners/oidc"
 )
 
-type GitHubAction struct{}
+var defaultAudience = []string{"nobody"}
 
-func NewGithubAction() *GitHubAction {
-	return &GitHubAction{}
+type GitHubAction struct {
+	githubToken *oidc.Token
+}
+
+func NewGithubAction(ctx context.Context) *GitHubAction {
+	// In order to ensure that we are running in a non-falsifiable environment we get the OIDC
+	// from Github. That allows us to read the workflow file path and runnner type. If that can't
+	// be done we fallback to reading the env vars directly.
+	client, err := oidc.NewGitHubClient(oidc.WithAudience(defaultAudience))
+	if err != nil {
+		return &GitHubAction{
+			githubToken: nil,
```

Comment:
do we have any logger that we could inject here so we can `debug` the fact that the github runner detection is unauthenticated for example?

Reply:
Runners are created at the point where we don't have a logger anywhere, at least I don't see it.